common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1002,17 +995,17 @@ bool ap_wlan_hal_dwpal::update_vap_credentials(
                    << " auth_type: " << auth_type << " encr_type: " << enc_type
                    << " network_key: " << bss_info_conf.network_key << " bss_type: " << bss_type;
 
-        // We cannot configure STAs yet
+        // We cannot configure STAs here
         if ((bss_info_conf.bss_type & WSC::eWscVendorExtSubelementBssType::BACKHAUL_STA) != 0) {
-            LOG(ERROR) << "Autoconfiguration: STAs are not yet supported";
+            LOG(ERROR) << "Autoconfiguration: STA cannot be configured here";
             return false;
         }
 
         // We only can use AP entries, skip STAs if ended up on one
         std::string entry_mode;
         while (hostapd_vap_iterator != hostapd_config_vaps.end() &&
                hostapd_config_get_value(hostapd_vap_iterator->second, "mode", entry_mode) &&
-               entry_mode.compare("ap") != 0) {
+               entry_mode == "ap") {
```

Comment:
Yes, good catch, I have not tested after changing the function calls to compare operators. Did not want to spend time on that since the review was still not finished.